### ISSUE TITLE
Stop Overriding Existing Image Tags

### ISSUE THREAD

**Issue by felicianotech** [circleci/circleci-images#153]
When this repo build, previously published Docker image tags are overridden. We only really want to do this in the following situations:
- latest tag
- alias tags (The Go `1.9` tag when the go `1.9.3` tag is created for example)
- security or other errors done within an image from our own doing
- the digest for a Docker image/tag changes upstream

**Comment by ryanwohara** [comment]
I think it's fine to constantly generate our images. Any tweaks to anything gets published every time guaranteed. You can still lock to a sha if you don't want an image to change. I'd really like to figure out a way to document all our image shas and version control them for customers to be well informed about what they're using.

**Comment by felicianotech** [comment]
Well that's the problem relying on SHAs. @smart-alek and I were already discussing documenting the Docker image tags in an automated fashion, which we did, but it wasn't easy because they change often.

These SHAs would change every couple of days. There's be so much to document and keep up-to-date.

**Comment by eric-hu** [comment]
I think it's worth checking whether upstream (official Docker) images will rewrite specific version tags like `1.9.3`, and not just natural group tags, like `1.9`.  If this is the case, it seems like an antipattern for us to pin `1.9.3` to a SHA when upstream changes what `1.9.3` refers to.  When I looked into Ruby, I got the impression that this might be happening.  I'm not sure they mutate tags though, so I think it just warrants further investigation.

A Docker bot updates [this repo](https://github.com/docker-library/official-images/tree/master/library) whenever an official image is added.  It might be helpful in answering the question of whether upstream images change tag references or not.

**Comment by felicianotech** [comment]
So I checked the Docker Library Go and MariaDB repos and SemVar tags are indeed being overriden. The Go 1.9.4 image was release over 10 days ago, but the tag was updated 2 days ago. Similar story with the MariaDB 10.2.13 image.

While I don't agree with the philosophy, clearly upstream is overwriting these tags which means we in turn should be doing so as well. This allows us to diverge as little as possible from upstream.
